### ISSUE TITLE
protecting figure with matplotlib widgets

### ISSUE THREAD

**Issue by pavoljuhas** [matplotlib/matplotlib#984]
Ubuntu Linux 10.04, matplotlib version: 0.99.1.1, backend GtkAgg;
probably the same in all later versions

There seems to be no way of protecting a GUI-only figure containing
matplotlib widgets.  If this figure becomes active, any pyplot command
will draw over the widgets.  As an example, this happens in the built-in
subplot_tool function where the plot command draws into the Reset box:

```
$ ipython -pylab
In [1]: subplot_tool()
# click any slider
In [2]: plot(range(2))
```

MATLAB deals with this issue by having a HandleVisibility and IntegerHandle
properties, which prevent a GUI figure from becoming the active one and protect
it from accidental changes.  It would be surely useful to have some similar
option of protecting GUI figures in matplotlib.

**Comment by petehuang** [comment]
#2624 linked to MEP27/#4143, just tossing a link here.

**Comment by QuLogic** [comment]
This seems to have been fixed since at least 2.0, as calling `subplot_tool` opens a (empty) figure which becomes active instead of it, so nothing is drawn on the Reset button.
